### PR TITLE
Skip some warning test on 3.10 to avoid MKL-related warning (and failure)

### PR DESCRIPTION
**Pull request by NicolasHug**
CC @malfet @atalman following up on our offline conversation, feel free to merge this PR or to submit a quick fix by pinning the numpy version as discussed.

### CODE REVIEW

File: test/smoke_test.py [hunk 1]
```diff
@@ -27,13 +27,15 @@ def smoke_test_torchvision_read_decode() -> None:
     if img_png.ndim != 3 or img_png.numel() < 100:
         raise RuntimeError(f"Unexpected shape of img_png: {img_png.shape}")
 
+
```
[the remark this answers is not in the document]
Comment (pmeier) [reply test/smoke_test.py:30]: Yeah, they come from #7359.

File: test/test_transforms.py [hunk 2]
```diff
@@ -2268,6 +2269,7 @@ def test_random_grayscale_with_grayscale_input():
 
 
 # TODO: remove in 0.17 when we can delete functional_pil.py and functional_tensor.py
+@pytest.mark.skipif(sys.version_info[:2] == (3, 10), reason="See #7372")
```
Comment (pmeier) [inline test/test_transforms.py:2272]: Can we make them xfails?

```suggestion
@pytest.mark.xfail(sys.version_info[:2] == (3, 10), reason="See #7372")
```

File: test/test_transforms_v2.py [hunk 3]
```diff
@@ -2087,6 +2088,7 @@ def test_sanitize_bounding_boxes_errors():
         transforms.SanitizeBoundingBox()(different_sizes)
 
 
+@pytest.mark.skipif(sys.version_info[:2] == (3, 10), reason="See #7372")
```
Comment (pmeier) [inline test/test_transforms_v2.py:2091]:
```suggestion
@pytest.mark.xfail(sys.version_info[:2] == (3, 10), reason="See #7372")
```

File: test/test_transforms_v2.py [hunk 4]
```diff
@@ -2120,6 +2122,7 @@ def test_warnings_v2_namespaces(import_statement, call_disable_warning):
     assert_run_python_script(textwrap.dedent(source))
 
 
+@pytest.mark.skipif(sys.version_info[:2] == (3, 10), reason="See #7372")
```
Comment (pmeier) [inline test/test_transforms_v2.py:2125]:
```suggestion
@pytest.mark.xfail(sys.version_info[:2] == (3, 10), reason="See #7372")
```

**Review by pmeier — APPROVED** [review]
Thanks!

**Review by atalman — APPROVED** [review]
LGTM

**Review by malfet — APPROVED** [review]
LGTM. By the way, did some digging: somehow those failures are triggered by pytest environment: running the same script outside of pytest did not result in the failure.

**Comment by pmeier** [comment]
It seems the issue is gone after #7189:

https://github.com/pytorch/vision/actions/runs/4308089284/jobs/7513938565#step:10:2643

I'll send a separate PR for the lint fixes. But otherwise, we can probably close this one?

**Comment by NicolasHug** [comment]
Thanks all. The [surfacing of the] issue is gone on `main` as @pmeier described just above so I'll close this one and open the same PR against `release/0.15` instead. (we could also cherry-pick https://github.com/pytorch/vision/pull/7189 into the release branch but that sounds riskier)

**Pull request closed without merging**